packages/autocannon-ui-frontend/src/components/RunOptions.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import { Grid, Container, TextField } from '@material-ui/core'
+import T from 'prop-types'
+
+import logo from '../assets/autocannon-banner.png'
+
+const useStyles = makeStyles(theme => ({
+  form: {
+    width: '100%',
+    marginTop: theme.spacing(1)
+  },
+  logo: {
+    maxWidth: '70%',
+    display: 'block',
+    margin: 'auto',
+    padding: '10px'
+  }
+}))
+
+export default function RunOptions(props) {
+  const classes = useStyles()
+
+  function onOptionChange(option, value) {
+    props.onOptionsChange({ ...props.options, [option]: value })
```

Comment:
for a race-condition-free way of mutating an existing state value

```suggestion
    props.onOptionsChange(o => ({...o, [option]: value }))
```